app/src/main/java/com/codingblocks/cbonlineapp/fragments/LeaderboardFragment.kt

```diff
@@ -0,0 +1,84 @@
+package com.codingblocks.cbonlineapp.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.codingblocks.cbonlineapp.R
+import com.codingblocks.cbonlineapp.adapters.LeaderboardListAdapter
+import com.codingblocks.cbonlineapp.database.AppDatabase
+import com.codingblocks.cbonlineapp.extensions.retrofitCallback
+import com.codingblocks.cbonlineapp.util.RUN_ATTEMPT_ID
+import com.codingblocks.onlineapi.Clients
+import com.codingblocks.onlineapi.models.Leaderboard
+import kotlinx.android.synthetic.main.fragment_leaderboard.*
+import org.jetbrains.anko.doAsync
+import kotlin.concurrent.thread
+
+class LeaderboardFragment : Fragment() {
+
+    lateinit var attemptId: String
+    lateinit var runId: String
+    private lateinit var leaderboardListAdapter: LeaderboardListAdapter
+    private var data = ArrayList<Leaderboard>()
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            attemptId = it.getString(RUN_ATTEMPT_ID)
+        }
+
+        getRunId()
+
+        leaderboardListAdapter = LeaderboardListAdapter(data)
+    }
+
+    private fun getRunId() {
```

Comment:
There is no need to get this id here can be directly passed from my courses activity see overview fragment

Reply:
What about this ?

Reply:
remove this database call it is not required and even if it is required move it to ViewModel